### DISCUSSION TITLE
odin doesn't compile project if the name of project folder is like `name.odin`

### DISCUSSION THREAD

**Discussion by jsomedon** [odin-lang/Odin#2596] [Ideas/Requests]
Say your project is in a folder `name.odin`, if you `odin run name.odin`, or if you `cd name.odin` then `odin run .` the compilation fails with errors like:

```
/path/name.odin(1:1) Syntax Error: Illegal character NUL
/path/name.odin(1:2) Syntax Error: Illegal character NUL
/path/name.odin(1:3) Syntax Error: Illegal character:  (0)
/path/name.odin(1:2) Syntax Error: Failed to parse file: /path/name; invalid token found in file
```

It looks like the compiler is treating the project folder as a single file(well, isn't everything file on unix?😄)

If odin could support folder names to be ending with `.odin`, then we could name our project as `whatever.odin` on like github, and I think that would be pretty cool. We see similar naming on zig and rust communities, like for rust, they have names like `project.rs`, for zig they have like `project.zig`.

**Comment by gingerBill** [comment]
As we discussed on the Discord, this is not a bug, per se, because directories are just files at the end of the day. Odin does look for files within a directory ending with `.odin`.
